### ISSUE TITLE
Very poor performance with YOLOv26 compared to previous versions

### ISSUE THREAD

**Issue by az45uhuj** [ultralytics/ultralytics#23356]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I tested YOLOv26 using the same setup and dataset as previous YOLO versions, but the performance is significantly worse (both accuracy and/or speed).

This feels like a regression rather than expected behavior.  
Is there a known issue, required config change, or recommended setup for YOLOv26?

Happy to provide logs or a minimal reproduction if needed

yolo11n.pt result
![Image](https://github.com/user-attachments/assets/3c0e2f4b-d1fd-4762-85a7-3fb03eeaab64)
yolo26n.pt result: 0: 640x512 (no detections), 145.1ms
![Image](https://github.com/user-attachments/assets/823da988-44f0-4f73-9e9e-f9c52e97cdcb)t

**Comment by UltralyticsAssistant** [comment]
👋 Hello @az45uhuj, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Thanks for sharing the screenshots and details—this does sound like something we should look into. This is an automated response 🤖, and an Ultralytics engineer will also assist soon.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community. 😄

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Are you using latest Ultralytics?

**Comment by Fiseldisel** [comment]
i have same issue, i compare yolo11 vs yolo26 with the same dataset and model.
i used cpu inference:
| Metric | yolo11 | yolo26 | Delta |
| :--- | :---: | :---: | :---: |
| **mAP50** | **0,6632** | 0,6614 | -0,0018 |
| **mAP50-95** | 0,4268 | **0,4313** | +0,0045 |
| **Precision** | **0,6413** | 0,6275 | -0,0138 |
| **Recall** | **0,5799** | 0,5733 | -0,0066 |
| **Latency (ms)** | 55,4503 | **52,1999** | -3,2504 |
| **FPS** | 18,0342 | **19,1571** | +1,1229 |1

**Comment by glenn-jocher** [comment]
@Fiseldisel your numbers look within run-to-run noise (and actually slightly better `mAP50-95` + lower latency), so that part doesn’t suggest a YOLO26 regression; the “no detections” case is usually a settings/version mismatch, so please share the exact `val`/`predict` commands you ran plus the text output of `yolo checks` (and if you recently upgraded in a notebook, restart the kernel once) so we can line up configs—YOLO26 changed to native end-to-end (no NMS) and other internals described in the [YOLO26 docs](https://docs.ultralytics.com/models/yolo26/).  

```bash
pip install -U ultralytics
yolo checks
yolo val model=yolo26n.pt data=your.yaml imgsz=640 device=cpu
yolo predict model=yolo26n.pt source=your.jpg imgsz=640 device=cpu conf=0.001
```

**Comment by Fiseldisel** [comment]
> [@Fiseldisel](https://github.com/Fiseldisel) your numbers look within run-to-run noise (and actually slightly better `mAP50-95` + lower latency), so that part doesn’t suggest a YOLO26 regression; the “no detections” case is usually a settings/version mismatch, so please share the exact `val`/`predict` commands you ran plus the text output of `yolo checks` (and if you recently upgraded in a notebook, restart the kernel once) so we can line up configs—YOLO26 changed to native end-to-end (no NMS) and other internals described in the [YOLO26 docs](https://docs.ultralytics.com/models/yolo26/).
> 
> pip install -U ultralytics
> yolo checks
> yolo val model=yolo26n.pt data=your.yaml imgsz=640 device=cpu
> yolo predict model=yolo26n.pt source=your.jpg imgsz=640 device=cpu conf=0.001

Thanks for the detailed explanation! I've updated to the latest ultralytics version and ran the checks as requested.
In my case, I don't actually have the 'no detections' issue. My dataset consists of very small objects (averaging around 20 pixels in area), which makes it quite challenging. The metrics for YOLO26 are very close to YOLO11, and the latency is indeed slightly better on my CPU.
Here are the logs from my environment:
### 1. YOLO Checks
```text
Ultralytics 8.4.7  Python-3.13.9 torch-2.9.1+cpu CPU (AMD Ryzen 5 5600 6-Core Processor)
Setup complete (12 CPUs, 15.9 GB RAM, 315.8/344.7 GB disk)
OS: Windows-11-10.0.26100-SP0
numpy                   2.2.6>=1.23.0
matplotlib              3.10.8>=3.3.0
opencv-python           4.13.0.90>=4.6.0
pillow                  11.1.0>=7.1.2
pyyaml                  6.0.3>=5.3.1
requests                2.32.3>=2.23.0
scipy                   1.16.3>=1.4.1
torch                   2.9.1<2.10,>=1.8.0
torch                   2.9.1!=2.4.0,<2.10,>=1.8.0; sys_platform == "win32"
torchvision             0.24.1>=0.9.0
psutil                  7.2.1>=5.8.0
polars                  1.37.1>=0.20.0
ultralytics-thop        2.0.18>=2.0.18
``` 
### 2. Validation Output (YOLO26n)
```text
Ultralytics 8.4.7  Python-3.13.9 torch-2.9.1+cpu CPU (AMD Ryzen 5 5600 6-Core Processor)
YOLO26n summary (fused): 122 layers, 2,375,031 parameters, 0 gradients, 5.2 GFLOPs
val: Fast image access  (ping: 0.00.0 ms, read: 91.78.6 MB/s, size: 8.2 KB)
val: Scanning C:\Users\FisliIL\Documents\scripts\scripts_dataset\11vs26\labels.cache... 757 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 757/757 226.8Mit/s 0.0s
Model: yolov26n_rgb_universal_v2_640.pt
Class     Images  Instances      Box(P          R      mAP50  mAP50-95)
all        757        757      0.627      0.573      0.604      0.371
Speed: 1.0ms preprocess, 45.3ms inference, 0.0ms loss, 0.2ms postprocess per image
```
### 3. Predict Output (conf=0.001)
```text
Ultralytics 8.4.7  Python-3.13.9 torch-2.9.1+cpu CPU (AMD Ryzen 5 5600 6-Core Processor)
YOLO26n summary (fused): 122 layers, 2,375,031 parameters, 0 gradients, 5.2 GFLOPs

image 1/1 C:\Users\FisliIL\Documents\scripts\scripts_dataset\11vs26\image.jpg: 640x640 7 class_0s, 61.6ms
Speed: 3.1ms preprocess, 61.6ms inference, 0.3ms postprocess per image at shape (1, 3, 640, 640)
```

**Comment by glenn-jocher** [comment]
Thanks for the logs—your YOLO26n `val`/`predict` output looks healthy (detections present and postprocess is essentially nil), so for your “~20px objects” case the main lever is usually resolution: try training/validating at a larger `imgsz` (e.g. 1024–1280) and, if you’re running inference on high-res imagery, consider tiled inference via the [SAHI tiled inference guide](https://docs.ultralytics.com/guides/sahi-tiled-inference/); for an apples-to-apples speed comparison across formats on CPU you can also use [Benchmark mode](https://docs.ultralytics.com/modes/benchmark/) and share the output.

```bash
yolo benchmark model=yolov26n_rgb_universal_v2_640.pt data=your.yaml imgsz=640 device=cpu
```

If you paste the exact `train` args you used for YOLO11 vs YOLO26 (plus your native image resolution), I can suggest the most important small-object settings to align the runs.

**Comment by Fiseldisel** [comment]
> Thanks for the logs—your YOLO26n `val`/`predict` output looks healthy (detections present and postprocess is essentially nil), so for your “~20px objects” case the main lever is usually resolution: try training/validating at a larger `imgsz` (e.g. 1024–1280) and, if you’re running inference on high-res imagery, consider tiled inference via the [SAHI tiled inference guide](https://docs.ultralytics.com/guides/sahi-tiled-inference/); for an apples-to-apples speed comparison across formats on CPU you can also use [Benchmark mode](https://docs.ultralytics.com/modes/benchmark/) and share the output.
> 
> yolo benchmark model=yolov26n_rgb_universal_v2_640.pt data=your.yaml imgsz=640 device=cpu
> If you paste the exact `train` args you used for YOLO11 vs YOLO26 (plus your native image resolution), I can suggest the most important small-object settings to align the runs.

Thanks for the feedback! I've ran the benchmark and gathered the training arguments. 

**Important context:** My native images are already **640x640**, so I am training and validating at their original resolution. The ~20px objects are roughly 4x5 pixels in size, which is quite challenging.

### 1. Benchmark Results (CPU)
YOLO26n shows better mAP and higher FPS in OpenVINO compared to YOLO11n.

**YOLO26n:**
| Format     | Size (MB) | mAP50-95 | Inference (ms) | FPS   |
|------------|-----------|----------|----------------|-------|
| PyTorch    | 5.2       | 0.3714   | 44.3           | 22.57 |
| OpenVINO   | 9.5       | 0.2434*  | 27.39          | 36.51 |

**YOLO11n:**
| Format     | Size (MB) | mAP50-95 | Inference (ms) | FPS   |
|------------|-----------|----------|----------------|-------|
| PyTorch    | 5.2       | 0.3620   | 41.42          | 24.14 |
| OpenVINO   | 10.2      | 0.3525   | 28.57          | 35.00 |
*\*Note: I noticed a significant mAP drop for YOLO26n in exported formats (0.3714 -> 0.2434), while YOLO11n remains stable.*
### 2. Training Configurations
I used different augmentation strategies for the two runs. Notably, for YOLO11n, I integrated **Albumentations** to handle blur and noise.

**YOLO11n (with Albumentations):**
- `albu_gaussnoise_p: 0.3` (std_range: [0.01, 0.035])
- `albu_defocus_p: 0.1` (radius: [3, 6])
- `albu_motionblur_p: 0.1`
- `scale: 0.5`, `fliplr: 0.5`, `close_mosaic: 15`
- `epochs: 150`, `batch: 0.9`, `imgsz=640`

**YOLO26n (Standard):**
- `scale: 0.7`, `degrees: 10`, `fliplr: 0.5`, `close_mosaic: 20`
- `epochs: 200`, `batch: 0.95`, `imgsz=640`

Other training parameters not mentioned here were left at their default values.

Since my native resolution is 640x640, SAHI might not be applicable here. Would you still recommend training at imgsz=1024 (upsampling) to help the model with these 4x5 pixel objects, or are there specific hyperparameter adjustments you'd suggest for such tiny detections?
I am particularly interested in maximizing inference speed on CPU.

**Comment by Y-T-G** [comment]
@Fiseldisel SAHI would still help.

And you can export with `end2end=False` for better mAP. You need to install the main branch before export:

```
pip install git+https://github.com/ultralytics/ultralytics
```

**Comment by Fiseldisel** [comment]
@Y-T-G , @glenn-jocher 
Thanks for the help! 

Exporting with `end2end=False` worked perfectly and fixed the mAP drop I was seeing in the exported formats. 

I will also look into experimenting with SAHI on my 640x640 images as you suggested to further improve the detection of these tiny objects. 

Appreciate the support!

**Comment by glenn-jocher** [comment]
Good to hear `end2end=False` fixed the exported-format mAP drop—this usually points to a runtime/export compatibility gap with the native end-to-end head, so for OpenVINO I’d keep exporting that way for now and focus your small-object gains on slicing/augmentation rather than changing heads; if your priority is max CPU throughput, OpenVINO INT8 is typically the next step—see the [Export mode docs](https://docs.ultralytics.com/modes/export/) and [OpenVINO integration guide](https://docs.ultralytics.com/integrations/openvino/) for calibration details.

```bash
yolo export model=yolo26n.pt format=openvino end2end=False int8=True data=your.yaml imgsz=640
```
